File: core/src/main/java/pl/mpakula/shorturl/core/ports/outgoing/UrlMappingRepository.java
package pl.mpakula.shorturl.core.ports.outgoing;

import lombok.NonNull;
import pl.mpakula.shorturl.core.model.UrlMapping;

import java.math.BigInteger;
import java.util.Optional;

public interface UrlMappingRepository {

    UrlMapping save(@NonNull UrlMapping urlMapping);

    Optional<UrlMapping> findFirstByOriginalUrl(String originalUrl);

    Optional<UrlMapping> findById(BigInteger id);

}

File: core/src/main/java/pl/mpakula/shorturl/core/ports/outgoing/UrlMappingRepository.kt
package pl.mpakula.shorturl.core.ports.outgoing

import pl.mpakula.shorturl.core.model.UrlMapping
import java.math.BigInteger
import java.util.*

interface UrlMappingRepository {
    fun save(urlMapping: UrlMapping): UrlMapping
    fun findFirstByOriginalUrl(originalUrl: String?): Optional<UrlMapping>
    fun findById(id: BigInteger?): Optional<UrlMapping>
}